oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/mqe/rt/MQEVisitorBase.java

```diff
@@ -0,0 +1,211 @@
+            return result;
+        }
+    }
+
+    @Override
+    public ExpressionResult visitFunction0OP(MQEParser.Function0OPContext ctx) {
+        int opType = ctx.function0().getStart().getType();
+        ExpressionResult expResult = visit(ctx.expression());
+        if (StringUtil.isNotEmpty(expResult.getError())) {
+            return expResult;
+        }
+        try {
+            return doFunction0Op(expResult, opType);
+        } catch (IllegalExpressionException e) {
+            ExpressionResult result = new ExpressionResult();
+            result.setType(ExpressionResultType.UNKNOWN);
+            result.setError(e.getMessage());
+            return result;
+        }
+    }
+
+    @Override
+    public ExpressionResult visitFunction1OP(MQEParser.Function1OPContext ctx) {
+        int opType = ctx.function1().getStart().getType();
+        ExpressionResult expResult = visit(ctx.expression());
+        if (StringUtil.isNotEmpty(expResult.getError())) {
+            return expResult;
+        }
+        try {
+            return doFunction1Op(expResult, opType, Integer.parseInt(ctx.parameter().INTEGER().getText()));
+        } catch (IllegalExpressionException e) {
+            ExpressionResult result = new ExpressionResult();
+            result.setType(ExpressionResultType.UNKNOWN);
+            result.setError(e.getMessage());
+            return result;
+        }
+    }
+
+    @Override
+    public ExpressionResult visitTopNOP(MQEParser.TopNOPContext ctx) {
+        return visit(ctx.metric());
+    }
+
+    @Override
+    public ExpressionResult visitRelablesOP(MQEParser.RelablesOPContext ctx) {
+        ExpressionResult result = visit(ctx.expression());
+        if (!result.isLabeledResult()) {
+            // Reserve the original result type
+            result.setError("The result of expression [" + ctx.expression().getText() + "] is not a labeled result.");
+            return result;
+        }
+
+        List<String> relabelList = Collections.emptyList();
+        String newLabelValue = ctx.label().labelValue().getText();
+        String labelValueTrim = newLabelValue.substring(1, newLabelValue.length() - 1);
+        if (StringUtil.isNotBlank(labelValueTrim)) {
+            relabelList = Arrays.asList(labelValueTrim.split(Const.COMMA));
+        }
+        List<MQEValues> mqeValuesList = result.getResults();
+
+        if (mqeValuesList.size() != relabelList.size()) {
+            // Reserve the original result type
+            result.setError("The number of relabels is not equal to the number of labels.");
+            return result;
+        }
+        // For now, we only have a single label named `label`
+        for (int i = 0; i < mqeValuesList.size(); i++) {
+            mqeValuesList.get(i).getMetric().getLabels().get(0).setValue(relabelList.get(i));
+        }
+
+        return result;
+    }
+
+    @Override
+    public abstract ExpressionResult visitMetric(MQEParser.MetricContext ctx);
```

Comment:
@kezhenxu94 From my reading/understanding, this is the extension point. 
MQE on Query API implementation would pick query DAOs to access data from the database.
Alerting kernel will provide another implementation on this, to read data from time window based metric data.